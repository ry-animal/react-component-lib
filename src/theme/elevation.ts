export const ELEVATION_LIGHT = {
  low: "0px 2px 8px RGBA(0, 0, 0, 0.04)",
  medium: "0px 2px 16px RGBA(0, 0, 0, 0.16)",
  high: "0px 2px 24px RGBA(0, 0, 0, 0.24)",
};

export const ELEVATION_DARK = {
  low: "inherit",
  medium: "inherit",
  high: "inherit",
};
