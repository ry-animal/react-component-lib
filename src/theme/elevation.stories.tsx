import styled from "styled-components";
import { Text } from "../components/Text";
import { ELEVATION_LIGHT } from "./elevation";

const BoxContainer = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.space.s24};
  flex-direction: row;
  margin-bottom: ${({ theme }) => theme.space.s8};
`;

interface BoxProps {
  item: string;
}
const Box = styled.div<BoxProps>`
  align-self: center;
  background-color: ${({ theme }) => theme.color.SURFACE_2};
  margin-right: ${({ theme }) => theme.space.s32};
  width: 300px;
  height: 300px;
  box-shadow: ${({ theme, item }) => {
    if (item === "low") {
      return theme.elevation.low;
    }
    if (item === "high") {
      return theme.elevation.high;
    }
    return theme.elevation.medium;
  }};
`;

export default {
  title: "Theme/Elevation",
  parameters: { options: { showPanel: false } },
};

const ElevationItem = ({ name, value }: { name: string; value: string }) => (
  <BoxContainer>
    <Box item={name} />
    <div>
      <Text>
        <b>Key:</b> {name}
      </Text>
      <Text>
        <b>Value:</b> {value}
      </Text>
    </div>
  </BoxContainer>
);

export const ElevationList = (): JSX.Element[] =>
  Object.entries(ELEVATION_LIGHT).map(([key, value]) => (
    <ElevationItem key={key} name={key} value={value} />
  ));
