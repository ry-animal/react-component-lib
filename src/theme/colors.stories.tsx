/* eslint-disable no-magic-numbers */
import * as React from "react";
import { withDesign } from "storybook-addon-designs";
import { Switch, Text } from "../";
import { PALETTE_DARK, PALETTE_LIGHT } from "./colors";

export default {
  title: "Theme/Colors",
  decorators: [withDesign],
};

const colorBlock = {
  marginBottom: 32,
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  color,
}: {
  itemKey: string;
  itemValue: string;
  color: string;
}) => {
  const renderStyles = {
    margin: 0,
    padding: "32px 16px 16px",
    backgroundColor: itemValue,
  } as React.CSSProperties;

  return (
    <div>
      <div style={renderStyles}>
        <Text text={itemKey} color={color} weight="700" reset />
        <Text
          text={itemValue}
          variant="Caption"
          style={{
            color,
            textTransform: "uppercase",
          }}
          reset
        />
      </div>
    </div>
  );
};

const getKey = (position: number) => Object.keys(PALETTE_LIGHT)[position];
const getColors = (isDark: boolean) => (isDark ? PALETTE_DARK : PALETTE_LIGHT);

interface ColorGroup {
  isDark?: boolean;
}

function Primary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Primary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(0)}
        itemValue={getColors(isDark).PRIMARY_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(1)}
        itemValue={getColors(isDark).PRIMARY_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(2)}
        itemValue={getColors(isDark).PRIMARY_3}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Secondary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Secondary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(3)}
        itemValue={getColors(isDark).SECONDARY_1}
        color="black"
      />
      <Item
        itemKey={getKey(4)}
        itemValue={getColors(isDark).SECONDARY_2}
        color="black"
      />
      <Item
        itemKey={getKey(5)}
        itemValue={getColors(isDark).SECONDARY_3}
        color="black"
      />
    </div>
  );
}

function Success({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Success"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(6)}
        itemValue={getColors(isDark).SUCCESS_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(7)}
        itemValue={getColors(isDark).SUCCESS_2}
        color="white"
      />
      <Item
        itemKey={getKey(8)}
        itemValue={getColors(isDark).SUCCESS_3}
        color="white"
      />
    </div>
  );
}

function Error({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Error" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(9)}
        itemValue={getColors(isDark).ERROR_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(10)}
        itemValue={getColors(isDark).ERROR_2}
        color="white"
      />
      <Item
        itemKey={getKey(11)}
        itemValue={getColors(isDark).ERROR_3}
        color="white"
      />
    </div>
  );
}

function Surface({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Surface"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(12)}
        itemValue={getColors(isDark).SURFACE_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(13)}
        itemValue={getColors(isDark).SURFACE_2}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(14)}
        itemValue={getColors(isDark).SURFACE_3}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(15)}
        itemValue={getColors(isDark).SURFACE_4}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Texto({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Text" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(16)}
        itemValue={getColors(isDark).TEXT_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(17)}
        itemValue={getColors(isDark).TEXT_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(18)}
        itemValue={getColors(isDark).TEXT_3}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(19)}
        itemValue={getColors(isDark).TEXT_4}
        color={isDark ? "black" : "white"}
      />
    </div>
  );
}

function Basic({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Basic" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(20)}
        itemValue={getColors(isDark).WHITE}
        color="black"
      />
      <Item
        itemKey={getKey(21)}
        itemValue={getColors(isDark).BLACK}
        color="white"
      />
    </div>
  );
}

export const LightThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_LIGHT.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary />
    <Secondary />
    <Success />
    <Error />
    <Surface />
    <Texto />
    <Basic />
  </div>
);
LightThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6766",
  },
};

export const DarkThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_DARK.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary isDark />
    <Secondary isDark />
    <Success isDark />
    <Error isDark />
    <Surface isDark />
    <Texto isDark />
    <Basic isDark />
  </div>
);

DarkThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6825",
  },
};

export const ThemeSwitcher = () => {
  const [isDark, setIsdark] = React.useState(false);
  const handleRadioChange = () => setIsdark(!isDark);

  return (
    <>
      <div style={colorBlock}>
        <Switch
          checked={isDark}
          label="Dark Mode"
          onChange={handleRadioChange}
        />
      </div>
      {isDark ? <DarkThemePalette /> : <LightThemePalette />}
    </>
  );
};
ThemeSwitcher.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10158%3A130155",
  },
};
