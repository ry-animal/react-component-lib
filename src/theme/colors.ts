import { ThemeColors, ThemeName } from "./theme.interface";

export const lightPaletteColors = {
  PRIMARY_1: "#000000",
  PRIMARY_2: "#5F5F5F",
  PRIMARY_3: "#EEEEEE",

  SECONDARY_1: "#FFF6DA",
  SECONDARY_2: "#FFE8A1",
  SECONDARY_3: "#FFD968",

  SUCCESS_1: "#F2FBF8",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const lightPaletteBase = {
  SURFACE_1: "#F9F9F9",
  SURFACE_2: "#FFFFFF",
  SURFACE_3: "#EEEEEE",
  SURFACE_4: "#DBDBDB",

  TEXT_1: "#151515",
  TEXT_2: "#404040",
  TEXT_3: "#5F5F5F",
  TEXT_4: "#A4A4A4",
};

export const darkPaletteColors = {
  PRIMARY_1: "#F9F9F9",
  PRIMARY_2: "#EEEEEE",
  PRIMARY_3: "#151515",

  SECONDARY_1: "#FFE8A1",
  SECONDARY_2: "#FFD968",
  SECONDARY_3: "#B38402",

  SUCCESS_1: "#19C68C",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const darkPaletteBase = {
  SURFACE_1: "#151515",
  SURFACE_2: "#212322",
  SURFACE_3: "#404040",
  SURFACE_4: "#5F5F5F",

  TEXT_1: "#FFFFFF",
  TEXT_2: "#EEEEEE",
  TEXT_3: "#DBDBDB",
  TEXT_4: "#C3C3C3",
};

export const palletteBase = {
  BLACK: "#000000",
  WHITE: "#ffffff",
};

export const PALETTE_LIGHT: ThemeColors = {
  ...lightPaletteColors,
  ...lightPaletteBase,
  ...palletteBase,
};

export const PALETTE_DARK: ThemeColors = {
  ...darkPaletteColors,
  ...darkPaletteBase,
  ...palletteBase,
};

export function getColorsByTheme(theme: ThemeName): ThemeColors {
  if (theme === ThemeName.DARK) {
    return PALETTE_DARK;
  }

  return PALETTE_LIGHT;
}
