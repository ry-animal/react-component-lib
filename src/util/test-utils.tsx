import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
