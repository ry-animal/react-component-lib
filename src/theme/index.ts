import { getColorsByTheme } from "./colors";
import { ELEVATION_DARK, ELEVATION_LIGHT } from "./elevation";
import { FONT, FontFamily, FontSize, FontWeight } from "./fonts";
import { GRADIENT } from "./gradient";
import { LAYERS } from "./layers";
import { MEDIA_WIDTHS } from "./media";
import { OPACITY } from "./opacity";
import { RADIUS } from "./radius";
import { SPACE } from "./space";
import { ThemeName } from "./theme.interface";

const sharedThemeKeys = {
  radius: {
    ...RADIUS,
  },
  space: {
    ...SPACE,
  },
  type: {
    ...FONT,
  },
  layer: {
    ...LAYERS,
  },
  media: {
    ...MEDIA_WIDTHS,
  },
  opacity: {
    ...OPACITY,
  },
  gradient: {
    ...GRADIENT,
  },
};

export const lightColors = getColorsByTheme(ThemeName.LIGHT);
export const darkColors = getColorsByTheme(ThemeName.DARK);

export const LIGHT_THEME = {
  name: ThemeName.LIGHT,
  ...sharedThemeKeys,
  color: {
    ...lightColors,
  },
  elevation: {
    ...ELEVATION_LIGHT,
  },
};

export const DARK_THEME = {
  name: ThemeName.DARK,
  ...sharedThemeKeys,
  color: {
    ...darkColors,
  },
  elevation: {
    ...ELEVATION_DARK,
  },
};

// Legacy
export { FontFamily, FontSize, FontWeight, MEDIA_WIDTHS, SPACE };
