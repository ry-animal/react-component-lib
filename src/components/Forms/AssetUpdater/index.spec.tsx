import userEvent from "@testing-library/user-event";
import { AssetUpdater, defaultAssetStrings } from ".";
import { SPACE } from "../../../theme";
import { render, screen } from "../../../util/test-utils";

const EXAMPLE_MIME_TYPES = [
  "image/gif",
  "image/jpeg",
  "image/png",
  "image/webp",
];

describe("<AssetUpdater />", () => {
  const bannerProps = {
    allowedMimeTypes: EXAMPLE_MIME_TYPES,
    maxFileSizeInMB: 100,
    label: "Upload a banner",
    onChange: jest.fn(),
    onError: jest.fn(),
  };
  const avatarProps = {
    allowedMimeTypes: EXAMPLE_MIME_TYPES,
    maxFileSizeInMB: 10,
    label: "Upload an avatar",
    onChange: jest.fn(),
    onError: jest.fn(),
  };
  const tileProps = {
    allowedMimeTypes: EXAMPLE_MIME_TYPES,
    maxFileSizeInMB: 10,
    label: "Upload a tile",
    onChange: jest.fn(),
    onError: jest.fn(),
  };
  const defaultProps = () => ({
    bannerProps,
    avatarProps,
  });

  const fullProps = () => ({
    ...defaultProps(),
    tileProps,
    bannerImg: "/banner.jpg",
    avatarImg: "/avatar.jpg",
    tileImg: "/tile.jpg",
    collectionName: "Collection name",
    isHex: true,
  });

  describe("rendering", () => {
    test("should render default props", () => {
      expect(
        render(<AssetUpdater {...defaultProps()} />).asFragment()
      ).toMatchSnapshot();
    });

    test("should render with full props", () => {
      expect(
        render(<AssetUpdater {...fullProps()} />).asFragment()
      ).toMatchSnapshot();
    });

    test("adds margin to banner img when rendering tile", () => {
      const { container, rerender } = render(
        <AssetUpdater {...defaultProps()} />
      );
      const bannerWrapper = container.firstChild?.firstChild;
      expect(bannerWrapper).not.toHaveStyleRule("margin-right", SPACE.s24);
      rerender(<AssetUpdater {...fullProps()} />);
      expect(bannerWrapper).toHaveStyleRule("margin-right", SPACE.s24);
    });

    test("renders correct number of elements", () => {
      const { container, rerender } = render(
        <AssetUpdater {...defaultProps()} />
      );
      const numChildrenNoTile = 2;
      const numChildrenWithTile = 3;

      const wrapper = container.firstChild;
      expect(wrapper?.childNodes.length).toBe(numChildrenNoTile);
      rerender(<AssetUpdater {...fullProps()} />);
      expect(wrapper?.childNodes.length).toBe(numChildrenWithTile);
    });

    test("renders optional collection name", () => {
      const collectionName = "Collection name";
      const { rerender } = render(
        <AssetUpdater {...fullProps()} collectionName={collectionName} />
      );

      const collectionTitle = screen.getByText(collectionName);
      expect(collectionTitle).toBeInTheDocument();
      rerender(<AssetUpdater {...fullProps()} collectionName={undefined} />);
      expect(collectionTitle).toHaveTextContent("");
    });
  });

  describe("event handling", () => {
    test("clicking banner delete clears banner img", () => {
      const props = defaultProps();
      render(<AssetUpdater {...props} bannerImg="banner.jpg" />);
      userEvent.click(
        screen.getByRole("button", {
          name: defaultAssetStrings.deleteBannerLabel,
        })
      );
      expect(props.bannerProps.onChange).toBeCalled();
    });
  });
});
