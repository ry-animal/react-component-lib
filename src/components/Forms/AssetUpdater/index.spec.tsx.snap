// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetUpdater /> rendering should render default props 1`] = `
<DocumentFragment>
  .c4 {
  cursor: pointer;
  border: none;
  border-radius: 10000px;
  box-shadow: 0px 2px 8px RGBA(0,0,0,0.04);
  pointer-events: auto;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  width: 40px;
  height: 40px;
  background-size: 20px auto;
  background-image: url([object Object]);
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(255,255,255,0.7);
  -webkit-backdrop-filter: blur(15px);
  backdrop-filter: blur(15px);
}

.c4:disabled {
  pointer-events: none;
  cursor: default;
}

.c4:hover {
  opacity: 100%;
  background-color: rgba(255,255,255,0.9);
}

.c9 {
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 96px;
  width: 96px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c10 {
  border-radius: 50%;
}

.c8 {
  position: relative;
  height: 104px;
  width: 104px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c8:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 100%;
  color: #151515;
}

.c5 {
  display: none;
  width: 0;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1 {
  width: 100%;
}

.c3 {
  background: #EEEEEE;
  background-size: cover;
  background-position: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 16px;
  height: 165px;
  width: 100%;
}

.c6 {
  position: absolute;
  bottom: 0;
  left: 32px;
  -webkit-transform: translateY(45%);
  -ms-transform: translateY(45%);
  transform: translateY(45%);
}

.c7 {
  position: relative;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:767px) {

}

@media (max-width:767px) {
  .c6 {
    left: 16px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        label="Upload a banner"
        role="application"
      >
        <div
          class="c3"
        >
          <button
            aria-label="Upload a banner"
            class="c4"
          />
        </div>
        <input
          accept="image/gif,image/jpeg,image/png,image/webp"
          aria-invalid="false"
          aria-label="File selector"
          class="c5"
          type="file"
        />
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c2"
        label="Upload an avatar"
        role="application"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            size="96"
          >
            <div
              class="c9 c10"
              size="96"
            />
          </div>
          <div
            class="c11"
          >
            <button
              aria-label="Upload an avatar"
              class="c4"
            />
          </div>
        </div>
        <input
          accept="image/gif,image/jpeg,image/png,image/webp"
          aria-invalid="false"
          aria-label="File selector"
          class="c5"
          type="file"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AssetUpdater /> rendering should render with full props 1`] = `
<DocumentFragment>
  .c4 {
  cursor: pointer;
  border: none;
  border-radius: 10000px;
  box-shadow: 0px 2px 8px RGBA(0,0,0,0.04);
  pointer-events: auto;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  width: 40px;
  height: 40px;
  background-size: 20px auto;
  background-image: url([object Object]);
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(255,255,255,0.7);
  -webkit-backdrop-filter: blur(15px);
  backdrop-filter: blur(15px);
}

.c4:disabled {
  pointer-events: none;
  cursor: default;
}

.c4:hover {
  opacity: 100%;
  background-color: rgba(255,255,255,0.9);
}

.c9 {
  position: relative;
  height: 104px;
  width: 104px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-clip-path: url(#hexPath);
  clip-path: url(#hexPath);
}

.c9:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c10 {
  background-image: url("/avatar.jpg");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 96px;
  width: 96px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c8 {
  position: absolute;
}

.c11 {
  -webkit-clip-path: url(#hexPath);
  clip-path: url(#hexPath);
}

.c18 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background-color: rgba(0,0,0,0.3);
  border-radius: 8px;
  padding: 16px;
  border-radius: 4px;
  padding: 12px;
}

.c21 {
  color: #ffffff;
  font-size: 32px;
  font-weight: 600;
  margin: 8px 16px;
  line-height: 38px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  word-break: break-word;
}

.c19 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.c17 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background: url(/tile.jpg) center no-repeat #F9F9F9;
  background-size: cover;
}

.c16 {
  background-color: #000000;
  border-radius: 8px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 475px;
  overflow: hidden;
  pointer-events: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  height: auto;
  min-height: 120px;
  border-radius: 4px;
}

.c16 .c20 {
  font-size: 12px;
  line-height: 12px;
  margin: 0;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 100%;
  color: #151515;
}

.c5 {
  display: none;
  width: 0;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1 {
  width: 100%;
  margin-right: 24px;
}

.c3 {
  background: url(/banner.jpg);
  background-size: cover;
  background-position: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 16px;
  height: 165px;
  width: 100%;
}

.c6 {
  position: absolute;
  bottom: 0;
  left: 32px;
  -webkit-transform: translateY(45%);
  -ms-transform: translateY(45%);
  transform: translateY(45%);
}

.c7 {
  position: relative;
}

.c12 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  position: absolute;
  right: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 {
  position: relative;
  width: 90px;
}

.c15 {
  background: #F9F9F9;
  border-radius: 6px;
  border: 3px solid #F9F9F9;
}

@media (max-width:767px) {
  .c1 {
    margin-right: 12px;
  }
}

@media (max-width:767px) {
  .c6 {
    left: 16px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        label="Upload a banner"
        role="application"
      >
        <div
          class="c3"
        >
          <button
            aria-label="Upload a banner"
            class="c4"
          />
          <button
            aria-label="Delete banner"
            class="c4"
          />
        </div>
        <input
          accept="image/gif,image/jpeg,image/png,image/webp"
          aria-invalid="false"
          aria-label="File selector"
          class="c5"
          type="file"
        />
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c2"
        label="Upload an avatar"
        role="application"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <svg
              fill="none"
              height="0"
              viewBox="0 0 50 50"
              width="0"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <clippath
                  clipPathUnits="objectBoundingBox"
                  id="hexPath"
                  transform="scale(0.02, 0.02)"
                >
                  <path
                    d="M26.2,0.7l19.2,11.1c0.7,0.4,1.2,1.2,1.2,2.1v22.2c0,0.9-0.5,1.7-1.2,2.1L26.2,49.3c-0.7,0.4-1.7,0.4-2.4,0L4.5,38.2c-0.7-0.4-1.2-1.2-1.2-2.1V13.9c0-0.9,0.5-1.7,1.2-2.1L23.8,0.7C24.5,0.3,25.5,0.3,26.2,0.7z"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="c9"
            size="96"
          >
            <div
              class="c10 c11"
              size="96"
            />
          </div>
          <div
            class="c12"
          >
            <button
              aria-label="Upload an avatar"
              class="c4"
            />
          </div>
        </div>
        <input
          accept="image/gif,image/jpeg,image/png,image/webp"
          aria-invalid="false"
          aria-label="File selector"
          class="c5"
          type="file"
        />
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <div
          class="c2"
          label="Upload a tile"
          role="application"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
              disabled=""
            >
              <div
                class="c17"
              />
              <div
                class="c18"
              >
                <div
                  class="c19"
                >
                  <h5
                    class="c20 c21"
                  >
                    Collection name
                  </h5>
                </div>
              </div>
            </div>
            <div
              class="c12"
            >
              <button
                aria-label="Upload a tile"
                class="c4"
              />
            </div>
          </div>
          <input
            accept="image/gif,image/jpeg,image/png,image/webp"
            aria-invalid="false"
            aria-label="File selector"
            class="c5"
            type="file"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
